File: app/LanguageQuiz/LanguageQuiz/Languini old Code/Languoid.m
//
//  Languoid.m
//  Languini
//
//  Created by Leo Thomas on 11/07/15.
//  Copyright (c) 2015 Coding Da Vinci. All rights reserved.
//

#import "Languoid.h"
#import "Country.h"
#import "Sentence.h"
#import <objc/runtime.h>


@implementation Languoid

@dynamic alternateNames;
@dynamic classification;
@dynamic dialects;
@dynamic iso6393;
@dynamic key;
@dynamic lanugageStatus;
@dynamic locationDescription;
@dynamic macroareaGl;
@dynamic name;
@dynamic population;
@dynamic populationNumeric;
@dynamic sentence;
@dynamic country;


// http://hesh.am/2013/01/transform-properties-of-an-nsobject-into-an-nsdictionary/
- (NSDictionary *)getExistingProperties {
    unsigned int count = 0;
    // Get a list of all properties in the class.
    objc_property_t *properties = class_copyPropertyList([self class], &count);
    
    NSMutableDictionary *dictionary = [[NSMutableDictionary alloc] initWithCapacity:count];
    
    for (int i = 0; i < count; i++) {
        NSString *key = [NSString stringWithUTF8String:property_getName(properties[i])];
        NSString *value = [self valueForKey:key];
        
        // Only add to the NSDictionary if it's not nil.
        if (value)
        dictionary[key] = value;
    }
    
    free(properties);
    
    return dictionary;
}

+ (NSString *)entityName {
    return @"Languiod";
}
@end

File: app/LanguageQuiz/Models/Languoid.swift
//
//  Languoid.swift
//  Languini
//
//  Created by Leo Thomas on 20/03/16.
//  Copyright © 2016 Coding Da Vinci. All rights reserved.
//

import Foundation
import CoreData

@objc(Languoid)
class Languoid: NSManagedObject {

    static let entityName = "Languiod"
    
    let detailUnusedAttributes = ["key", "macroareaGl", "populationNumeric"]

    
    func getCountryString() -> String {
        var resultString = ""
        guard let countriesArray = country?.allObjects as? [Country] else {
            return resultString
        }
        for (index, country) in countriesArray.enumerate() {
            resultString += country.name ?? ""
            if index < countriesArray.count-1 {
                resultString += ", "
            }
        }
        return resultString
    }
    
    func getDetailAttributes() -> [String] {
        var results = [String]()
        let entity = self.entity
        let attributes = entity.attributesByName
        for attribute in attributes{
            let attributeKey = attribute.0
            if let _ = self.valueForKey(attributeKey) {
                if !detailUnusedAttributes.contains(attributeKey){
                    if let value = self.valueForKey(attributeKey) {
                        print(value)
                        results.append(attributeKey)
                    }
                }
            }
        }
        return results
    }
}
